expression/builtin_json.go

```diff
@@ -740,6 +742,78 @@ func jsonModify(ctx sessionctx.Context, args []Expression, row chunk.Row, mt typ
 	return res, false, nil
 }
 
+type jsonMemberOfFunctionClass struct {
+	baseFunctionClass
+}
+
+type builtinJSONMemberOfSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinJSONMemberOfSig) Clone() builtinFunc {
+	newSig := &builtinJSONMemberOfSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+func (c *jsonMemberOfFunctionClass) verifyArgs(args []Expression) error {
+	if err := c.baseFunctionClass.verifyArgs(args); err != nil {
+		return err
+	}
+	if evalType := args[1].GetType().EvalType(); evalType != types.ETJson && evalType != types.ETString {
+		return types.ErrInvalidJSONData.GenWithStackByArgs(2, "member of")
+	}
+	return nil
+}
+
+func (c *jsonMemberOfFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+
+	argTps := []types.EvalType{args[0].GetType().EvalType(), types.ETJson}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETInt, argTps...)
+	if err != nil {
+		return nil, err
+	}
+	sig := &builtinJSONMemberOfSig{bf}
+	return sig, nil
+}
+
+func (b *builtinJSONMemberOfSig) evalInt(row chunk.Row) (res int64, isNull bool, err error) {
+	var target types.BinaryJSON
+	if b.args[0].GetType().EvalType() != types.ETJson {
+		eval, err := b.args[0].Eval(row)
+		if err != nil || eval.IsNull() {
+			return 0, eval.IsNull(), err
+		}
+		target = types.CreateBinaryJSON(eval.GetValue())
```

Comment:
At least, should we use `CreateBinaryJSONWithCheck`? For unexpected type (`[]byte`, `Enum`...), `CreateBinaryJSON` will panic.

Reply:
Another concern is about the string charset. In some situation, we may want to get an `Opaque`, but `eval.GetValue()` will never give us.

Maybe, we could auto convert the type in `(*baseFuncDesc).TypeInfer`? Then it could use the `castAsJSONFunctionClass` to convert from other types to json.